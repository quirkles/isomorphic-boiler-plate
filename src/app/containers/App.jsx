import React from 'react';
import propTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';

const App = ({ route }) => (
  <div>
    <ul>
      <li>
        <NavLink to="/home">home</NavLink>
      </li>
      <li>
        <NavLink to="/list">list</NavLink>
      </li>
      <li>
        <NavLink to="/signup">Sign up</NavLink>
      </li>
    </ul>
    <div>{renderRoutes(route.routes)}</div>
  </div>
);

App.propTypes = {
  route: propTypes.shape({
    routes: propTypes.array.isRequired,
  }).isRequired,
};

export default App;
