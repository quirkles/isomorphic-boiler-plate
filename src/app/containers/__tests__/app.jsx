import React from 'react';
import { mount } from 'enzyme';

import App from '../App';

const Component = () => <div className="component" />;

describe('App container', () => {
  it('Renders with route and matches snapshot', () => {
    let wrapper;
    const route = {
      routes: [
        {
          path: '/home',
          component: Component,
        },
      ],
    };
    expect(() => {
      wrapper = mount(<App route={route} />);
    }).not.toThrow();
    expect(wrapper).toMatchSnapshot();
  });
});
