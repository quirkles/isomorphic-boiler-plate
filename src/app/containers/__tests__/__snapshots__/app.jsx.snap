// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container Renders with route and matches snapshot 1`] = `
<App
  route={
    Object {
      "routes": Array [
        Object {
          "component": [Function],
          "path": "/home",
        },
      ],
    }
  }
>
  <div>
    <ul>
      <li>
        <NavLink
          to="/home"
        >
          <a
            href="/home"
          >
            home
          </a>
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/list"
        >
          <a
            href="/list"
          >
            list
          </a>
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/signup"
        >
          <a
            href="/signup"
          >
            Sign up
          </a>
        </NavLink>
      </li>
    </ul>
    <div>
      <div>
        [{"path":"/home"}]
      </div>
    </div>
  </div>
</App>
`;
